louve_addons/edi_purchase_base/models/supplier_price_list.py

```diff
@@ -14,9 +14,47 @@ class SupplierPriceList(models.Model):
     supplier_id = fields.Many2one(comodel_name="res.partner", string="EDI Supplier",
                                   domain="[('is_edi', '=', True), ('supplier', '=', True)]", readonly=True,
                                   required=True)
-    product_tmpl_id = fields.Many2one(comodel_name="product.template", string="Product")
+    product_tmpl_id = fields.Many2one(comodel_name="product.template", string="Product", ondelete='set null')
     product_name = fields.Char(string="Product name", readonly=True, required=True)
     supplier_code = fields.Char(string="Supplier code", readonly=True, required=True)
     price = fields.Float('Price', digits_compute=dp.get_precision('Product Price'), readonly=True, required=True,
                          help="The price HT to purchase a product")
     apply_date = fields.Date(string="Apply date", readonly=True, required=True)
+
+    @api.multi
+    def button_create_product(self):
+        self.ensure_one()
+        # create new product
+        product_tmpl_id = self.env['product.template'].create({'name': self.product_name,
+                                                               'sale_ok': True,
+                                                               'purchase_ok': True,
+                                                               'type': 'product'
+                                                               })
```

Comment:
```suggestion
        product_tmpl_id = self.env['product.template'].create({
            'name': self.product_name,
            'sale_ok': True,
            'purchase_ok': True,
            'type': 'product',
        })
```

In the future please use this code-style